apps/wallet-ui/src/components/TransactionToastProvider.tsx

```diff
@@ -0,0 +1,127 @@
+"use client";
+import { useToast } from "@/components/ui/use-toast";
+import { useQuery } from "@tanstack/react-query";
+import { CheckIcon, Loader2, OctagonXIcon } from "lucide-react";
+import Link from "next/link";
+import { useCallback, useMemo, useSyncExternalStore } from "react";
+import { type Hex, ZERO_ADDRESS, defineChain, waitForReceipt } from "thirdweb";
+import { useActiveAccount } from "thirdweb/react";
+import { getTransactionStore } from "thirdweb/transaction";
+import { getChain } from "../lib/chains";
+import { client } from "../lib/client";
+import { revalidate } from "../lib/revalidate";
+import { ToastAction } from "./ui/toast";
+import { Toaster } from "./ui/toaster";
+
+const transactionQueue = new Set();
+export function TransactionToastProvider() {
+  const account = useActiveAccount();
+  const { toast } = useToast();
+  const store = useMemo(() => {
+    return getTransactionStore(account?.address || ZERO_ADDRESS);
+  }, [account?.address]);
+
+  const snapshot = useSyncExternalStore(
+    store.subscribe,
+    store.getValue,
+    store.getValue,
+  );
```

Comment:
we don't have a better way to access this?

Reply:
I thought the whole point of the transaction store was to use this pattern?

Reply:
alright then (I don't remember, but possible)